ballerina/service/backend/main.bal

```diff
@@ -0,0 +1,121 @@
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied. See the License for the
+// specific language governing permissions and limitations
+// under the License.
+// These lines describe the conditions under which the software is distributed and disclaim any warranties or conditions.
+
+import ballerina/file;
+
+import ballerina/http;
+
+import ballerina/io;
+
+import ballerina/jballerina.java;
+
+import ballerina/regex;
+
+
+import ballerinax/googleapis.sheets as gsheets;
+
+type Auth record {|
+    string clientId;
+    string clientSecret;
+    string refreshToken;
+    string refreshUrl;
+
+|};
+
+configurable string pdfFilePath = ?;
+configurable string fontFilePath = ?;
+configurable int port = 9090;
+configurable string spreadsheetId = ?;
+configurable Auth auth = ?;
+
+const PDF_EXTENSION = ".pdf";
+const OUTPUT_DIRECTORY = "outputs/";
+const FILE_NAME = "certificate.pdf";
+const NAME_COLUMN = "C";
+const SUCCESS_CODE = 200;
+const ERROR_CODE = 400;
+const CONTENT_TYPE = "application/pdf";
+const CONTENT_DISPOSITION = "inline; filename='certificate.pdf'";
+
+gsheets:Client spreadsheetClient = check new ({
+    auth: auth
+});
+
+string filePath = "";
+isolated function generatePdf(handle pdfGenerator) = @java:Method {
+    'class: "org.PDFCreator.PDFGenerator",
+    name: "pdf"
+} external;
+
+isolated function generateParams(handle inputFilePath, handle replacement, handle fontType, int fontsize, int centerX, int centerY, handle fontFilePath, handle outputFileName) returns handle = @java:Method {
+    'class: "org.PDFCreator.PDFGenerationParams",
+    name: "createInstance"
+} external;
+
+public function certificateGeneration(string inputFilePath, string fontFilePath, string checkID, string sheetName) returns error? {
+    gsheets:Column col = check spreadsheetClient->getColumn(spreadsheetId, sheetName, NAME_COLUMN);
+    int i = 1;
+    while i < col.values.length() {
+        gsheets:Row row = check spreadsheetClient->getRow(spreadsheetId, sheetName, i);
+        if row.values[3].toString() == checkID {
+            string replacement = col.values[i].toString();
+            string fileName = replacement + PDF_EXTENSION;
+            filePath = OUTPUT_DIRECTORY + fileName;
+            int fontsize = check int:fromString(row.values[7].toString());
+            int centerX = check int:fromString(row.values[4].toString());
+            int centerY = check int:fromString(row.values[5].toString());
+            handle javastrName = java:fromString(replacement);
+            handle javafontType = java:fromString(row.values[6].toString());
+            handle javafontPath = java:fromString(fontFilePath);
+            handle pdfpath = java:fromString(inputFilePath);
+            handle javaOurputfileName = java:fromString(fileName);
+            handle pdfData = generateParams(pdfpath, javastrName, javafontType, fontsize, centerX, centerY, javafontPath, javaOurputfileName);
```

Comment:
I still recommend using Ballerina `string` values instead of using `handle` values on the Ballerina side. As a practice, reduce the usage of `handle` values, unless it is absolutely necessary.

Reply:
Since I use pdfbox libary to handle pdfs ,I cannot use BString as input arguments to those builtin functions.